plutus-pab/src/Plutus/PAB/Core/ContractInstance/BlockchainEnv.hs

```diff
@@ -57,13 +58,21 @@ startNodeClient socket mode slotConfig networkId instancesState = do
     case mode of
       MockNode ->
         void $ MockClient.runChainSync socket slotConfig
-            (\block slot -> STM.atomically $ processMockBlock instancesState env block slot)
+            (\block slot -> STM.atomically $ handleSyncAction $ processMockBlock instancesState env block slot)
       AlonzoNode -> do
           let resumePoints = []
           void $ Client.runChainSync socket slotConfig networkId resumePoints
-            (\block slot -> STM.atomically $ processChainSyncEvent env block slot)
+            (\block slot -> STM.atomically $ handleSyncAction $ processChainSyncEvent env block slot)
     pure env
 
+-- | Deal with errors. For now, we deal with them by simply failing the entire
+-- application.
+handleSyncAction :: STM (Either SyncActionFailure ()) -> STM ()
```

Comment:
I'd prefer to do the error handling outside of `STM`, something like

```haskell
handleSyncAction :: STM (Either SyncActionFailure ()) -> IO ()
handleSyncAction action = STM.atomically action >>= either (error . show) pure
```

Just seems a bit cleaner.
Other than that, using `error` here is fine for now.